### PR TITLE
STK-1409: add watch service update

### PR DESCRIPTION
**Pull request by garethj2**
This enables a developer to manually trigger a watch update without a poll request.

It's useful if you've just made a server side call and you want to also update any other changes in the UI without triggering a poll.

I know WebSockets would obviously solve this problem better. Stil this is useful for now.

### CODE REVIEW

File: spec/client/watches/ApiSubject.spec.ts [hunk 1]
```diff
@@ -350,6 +360,63 @@ describe('ApiSubject', function (): void {
 			expect(cb).toHaveBeenCalledWith(event)
 		})
 
+		it('does not invoke a callback is the new data is older than the current', async () => {
```
Comment (riccardoleder) [inline spec/client/watches/ApiSubject.spec.ts]:
```suggestion
		it('does not invoke a callback if the new data is older than the current', async () => {
```

File: spec/client/watches/ApiSubject.spec.ts [hunk 2]
```diff
@@ -350,6 +360,63 @@ describe('ApiSubject', function (): void {
...
+				}),
+			]
+
+			asMock(apis.poll).mockResolvedValue(changed)
+			await subject.poll()
+
+			expect(cb).not.toHaveBeenCalled()
+		})
+
+		it('invokes a callback is the new data is newer than the current', async () => {
```
Comment (riccardoleder) [inline spec/client/watches/ApiSubject.spec.ts]:
```suggestion
		it('invokes a callback if the new data is newer than the current', async () => {
```

File: spec/client/watches/ApiSubject.spec.ts (hunk 1 above)
Comment (riccardoleder) [reply spec/client/watches/ApiSubject.spec.ts]: @garethj2 should this say `older than or equally old to the current`? 
I'm guessing since this is based on `HDict.isNewer` this would be the case.

Or as an alternative could it be worth to also test that edge case? (Assuming we don't want the callback to be invoked with the exact same data)

**Review by riccardoleder — APPROVED** [review]
Looks good to me

**Review by riccardoleder — APPROVED** [review]
